### ISSUE TITLE
[java] MethodReturnsInternalArray: Various false negatives with local aliases and conditional expressions

### ISSUE THREAD

**Issue by MarkLee131** [pmd/pmd#6901]
**Affects PMD Version:** 7.26.0

**Rule:** [MethodReturnsInternalArray](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#methodreturnsinternalarray)

**Description:**

`MethodReturnsInternalArray` only reports `return <field>` syntactically. The same array returned through a local alias, a conditional expression, branch assignments, or a generic wrapper is not reported. Behaviourally identical escapes, so these are false negatives.

**Code Sample demonstrating the issue:**

```java
public class ArrDirect {             // reported
    private int[] data = new int[4];

    public int[] getData() {
        return data;
    }
}
```

```java
public class ArrAlias {              // NOT reported
    private int[] data = new int[4];

    public int[] getData() {
        int[] result = data;
        return result;
    }
}
```

```java
public class ArrConditional {        // NOT reported
    private int[] data = new int[4];
    private boolean flag;

    public int[] getData() {
        return flag ? data : new int[0];
    }
}
```

```java
public class ArrBranches {           // NOT reported
    private int[] data = new int[4];
    private boolean flag;

    public int[] getData() {
        int[] result;
        if (flag) {
            result = data;
        } else {
            result = new int[0];
        }
        return result;
    }
}
```

```java
public class ArrVariant {            // NOT reported (original reproducer)
    private int[] data = new int[4];

    public int[] getData() {
        return java.util.Collections.singletonList(data).get(0);
    }
}
```

**Expected outcome:**

PMD should report a `MethodReturnsInternalArray` violation in each class above; in 7.26.0 only `ArrDirect` is reported. Returning a defensive copy such as `data.clone()` or `Arrays.copyOf(...)` remains unreported.

**Running PMD through:** CLI

These examples match the test cases added in #6940.

**Comment by fudianchn** [comment]
@adangel As discussed in #6940, I can't edit this issue. Proposed title and description for a maintainer to copy:

**Proposed title:**

[java] MethodReturnsInternalArray: Various false negatives with local aliases and conditional expressions

**Proposed description:**

**Affects PMD Version:** 7.26.0

**Rule:** MethodReturnsInternalArray
https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#methodreturnsinternalarray

**Description:**

`MethodReturnsInternalArray` only reports `return <field>` syntactically. The same array returned through a local alias, a conditional expression, branch assignments, or a generic wrapper is not reported. Behaviourally identical escapes, so these are false negatives.

**Code Sample demonstrating the issue:**

```java
public class ArrDirect {             // reported
    private int[] data = new int[4];

    public int[] getData() {
        return data;
    }
}
```

```java
public class ArrAlias {              // NOT reported
    private int[] data = new int[4];

    public int[] getData() {
        int[] result = data;
        return result;
    }
}
```

```java
public class ArrConditional {        // NOT reported
    private int[] data = new int[4];
    private boolean flag;

    public int[] getData() {
        return flag ? data : new int[0];
    }
}
```

```java
public class ArrBranches {           // NOT reported
    private int[] data = new int[4];
    private boolean flag;

    public int[] getData() {
        int[] result;
        if (flag) {
            result = data;
        } else {
            result = new int[0];
        }
        return result;
    }
}
```

```java
public class ArrVariant {            // NOT reported (original reproducer)
    private int[] data = new int[4];

    public int[] getData() {
        return java.util.Collections.singletonList(data).get(0);
    }
}
```

**Expected outcome:**

PMD should report a `MethodReturnsInternalArray` violation in each class above; in 7.26.0 only `ArrDirect` is reported. Returning a defensive copy such as `data.clone()` or `Arrays.copyOf(...)` remains unreported.

**Running PMD through:** CLI

These examples match the test cases added in #6940.

**Comment by adangel** [comment]
@fudianchn Thanks for the updated description!
